### PR TITLE
Replace tmpdir fixture with tmp_path in tests (#3551)

### PR DESCRIPTION
**Pull request by bruziev**
## Related issue number
#3551

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,26 +1,13 @@
...
-import tempfile
 
 import pytest
 import trustme
 
 pytest_plugins = ['aiohttp.pytest_plugin', 'pytester']
 
 
-@pytest.fixture
-def shorttmpdir():
```
Comment (asvetlov) [inline tests/conftest.py:15]: IIRC thee fixture exists to satisfy cases when `path is too long` was reported by CI.
If travis/appveyor are happy without the fixture -- that's fine to me.
We can reinvent it later if needed.

File: tests/test_connector.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def ssl_key():
 
 
 @pytest.fixture
-def unix_sockname(shorttmpdir):
-    sock_path = shorttmpdir / 'socket.sock'
+def unix_sockname(tmpdir):
```
Comment (asvetlov) [inline tests/test_connector.py:46]: `tmpdir` -> `tmp_path`

File: tests/conftest.py (hunk 1 above)
Comment (blueyed) [reply tests/conftest.py:15]: If this needs some fix later again it should be considered to get fixed in pytest also.
(Some quick thought: maybe the (temp) base dir could be shortened on Travis if that causes trouble, e.g. by exporting TMPDIR or something alike.)

**Comment by asvetlov** [comment]
thanks!

File: tests/test_connector.py (hunk 2 above)
Comment (blueyed) [reply tests/test_connector.py:46]: This was not done/changed.

Comment (asvetlov) [reply tests/test_connector.py:46]: Yes, I understand.
A task for the next PR?

Comment (blueyed) [reply tests/test_connector.py:46]: => https://github.com/aio-libs/aiohttp/pull/3555

**Pull request merged**